src/aosm/azext_aosm/generate_nsd/nsd_generator.py

```diff
@@ -46,20 +55,42 @@ class NSDGenerator:
       be deployed by the NSDV
     """
 
-    def __init__(self, config: NSConfiguration):
+    def __init__(self, api_clients: ApiClients, config: NSConfiguration):
         self.config = config
         self.nsd_bicep_template_name = NSD_DEFINITION_BICEP_SOURCE_TEMPLATE
         self.nf_bicep_template_name = NF_TEMPLATE_BICEP_FILE
         self.nsd_bicep_output_name = NSD_DEFINITION_BICEP_FILE
 
         self.build_folder_name = self.config.build_output_folder_name
+        nfdv = self._get_nfdv(config, api_clients)
+        print("Finding the deploy parameters of the NFDV resource")
+        if not nfdv.deploy_parameters:
+            raise NotImplementedError(
+                "NFDV has no deploy parameters, cannot generate NSD."
+            )
+        self.deploy_parameters: str = nfdv.deploy_parameters
+
+    def _get_nfdv(
+        self, config: NSConfiguration, api_clients
+    ) -> NetworkFunctionDefinitionVersion:
+        """Get the existing NFDV resource object."""
+        print(
+            "Reading existing NFDV resource object "
+            f"{config.network_function_definition_version_name} from group "
+            f"{config.network_function_definition_group_name}"
+        )
+        nfdv_object = api_clients.aosm_client.network_function_definition_versions.get(
+            resource_group_name=config.publisher_resource_group_name,
+            publisher_name=config.publisher_name,
+            network_function_definition_group_name=config.network_function_definition_group_name,
+            network_function_definition_version_name=config.network_function_definition_version_name,
+        )
+        return nfdv_object
 
-    def generate_nsd(self, deploy_parameters) -> None:
+    def generate_nsd(self) -> None:
         """Generate a NSD templates which includes an Artifact Manifest, NFDV and NF templates."""
         logger.info(f"Generate NSD bicep templates")
 
-        self.deploy_parameters = deploy_parameters
```

Comment:
Ah - glad you moved this.  It was annoying me that it wasn't in `__init__()` yesterday.